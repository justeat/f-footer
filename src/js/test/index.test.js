import TestUtils from 'js-test-buddy';
import { collapseFooterPanels, resizeInit } from '../';

beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
        value: jest.fn(() => ({ matches: true }))
    });
});

describe('on page load', () => {

    it('footer panels are collapsed', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <div class="c-footer-panel" data-panel-collapsible="">
                <h2>
                    Header text
                </h2>
            </div>
        `);

        // Act
        collapseFooterPanels();
        const html = TestUtils.getBodyHtml();

        // Assert
        expect(html).toMatchSnapshot();
    });

    describe('at narrow width', () => {

        beforeEach(() => {
            // Force matchMedia in tabindexResize to fail
            Object.defineProperty(window, 'matchMedia', {
                value: jest.fn(() => ({ matches: false }))
            });
        });

        it('tabindex should not be removed from footer panel headings', () => {
            // Arrange
            TestUtils.setBodyHtml(`
                <div class="c-footer-panel" data-panel-collapsible="">
                    <h2 data-footer-panel-heading tabindex="0">
                        Header text
                    </h2>
                </div>
            `);

            // Act
            resizeInit();
            const html = TestUtils.getBodyHtml();

            // Assert
            expect(html).toMatchSnapshot();
        });

        it('tabindex should be added to footer panel headings if it is not present', () => {
            // Arrange
            TestUtils.setBodyHtml(`
                <div class="c-footer-panel" data-panel-collapsible="">
                    <h2 data-footer-panel-heading>
                        Header text
                    </h2>
                </div>
            `);

            // Act
            resizeInit();
            const html = TestUtils.getBodyHtml();

            // Assert
            expect(html).toMatchSnapshot();
        });
    });

    describe('at normal width', () => {

        it('tabindex should be removed from footer panel headings', () => {
            // Arrange
            TestUtils.setBodyHtml(`
                <div class="c-footer-panel" data-panel-collapsible="">
                    <h2 data-footer-panel-heading tabindex="0">
                        Header text
                    </h2>
                </div>
            `);

            // Act
            resizeInit();
            const html = TestUtils.getBodyHtml();

            // Assert
            expect(html).toMatchSnapshot();
        });
    });
});
