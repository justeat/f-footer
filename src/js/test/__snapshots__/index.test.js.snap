// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on page load at narrow width tabindex should be added to footer panel headings if it is not present 1`] = `
"
                <div class=\\"c-footer-panel\\" data-panel-collapsible=\\"\\">
                    <h2 data-footer-panel-heading=\\"\\" tabindex=\\"0\\">
                        Header text
                    </h2>
                </div>
            <div class=\\"c-screen-sizer\\"></div>"
`;

exports[`on page load at narrow width tabindex should not be removed from footer panel headings 1`] = `
"
                <div class=\\"c-footer-panel\\" data-panel-collapsible=\\"\\">
                    <h2 data-footer-panel-heading=\\"\\" tabindex=\\"0\\">
                        Header text
                    </h2>
                </div>
            <div class=\\"c-screen-sizer\\"></div>"
`;

exports[`on page load at normal width tabindex should be removed from footer panel headings 1`] = `
"
                <div class=\\"c-footer-panel\\" data-panel-collapsible=\\"\\">
                    <h2 data-footer-panel-heading=\\"\\">
                        Header text
                    </h2>
                </div>
            <div class=\\"c-screen-sizer\\"></div>"
`;

exports[`on page load footer panels are collapsed 1`] = `
"
            <div class=\\"c-footer-panel is-collapsed\\" data-panel-collapsible=\\"\\">
                <h2>
                    Header text
                </h2>
            </div>
        "
`;
